pkg/location/extensions/extensions.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"context"
+	"fmt"
+)
+
+type Reference interface {
+	fmt.Stringer
+	Type() string
+	Validate() error
+}
```

Comment:
I am wondering if this should have `ID()` method that returns string ?

Reply:
Sure, what would be the semantics? Is the idea that `ID()` would return a nearly total string value that would round-trip through parsing if possible - and the `fmt.Stringer.String()` would return a string that's potentially more optimized for display/log/error purposes?

@martinmaly had earlier questions related to round-tripping that could be related. Also this would help with "-" locations... As a Content.ID() it works great for round-tripping, but as a Stringer.String() a single dash is a bit of miserable display name.